### PR TITLE
OAuth API Merge

### PR DESCRIPTION
**Pull request by jakubvul**
## Description:
This PR does the following:

- Merges the existing OAuth 2.0 and OIDC API pages into [one single page](https://github.com/okta/okta.github.io/pull/1742/files#diff-cc7368d5ba3efa5e754031db96a48e49).
- Removes the existing OAuth 2.0 API page.
- Moves all of the Authorization Server-related endpoint documentation into a [separate page](https://github.com/okta/okta.github.io/pull/1742/files#diff-50856496c2bcee5e7bd8994ef1e76f0b) under the Management APIs called "Authorization Servers"
- Changes all existing links to the old OAuth 2.0 API documentation page to equivalent links one of the new API pages.
- Adds "Authorization Server" link under Management API section of left-side nav bar.

Has already been reviewed in individual PRs by @nbarbettini and @yuliu-okta. This is the final PR that merges all of the individual PRs.

**Please contact me before you leave any comments, since we've learned that GitHub is very bad at handling PRs of this size.**

### Resolves:

* [OKTA-152508](https://oktainc.atlassian.net/browse/OKTA-152508)
* [OKTA-152793](https://oktainc.atlassian.net/browse/OKTA-152793)

### CODE REVIEW

File: _source/_docs/api/resources/oidc.md [hunk 1]
```diff
@@ -1,189 +1,387 @@
 ---
 layout: docs_page
-title: OpenID Connect
+title: OpenID Connect & OAuth 2.0 API
+redirect_from: "/docs/api/resources/oauth2.html"
```
Comment (jmelberg-okta) [inline _source/_docs/api/resources/oidc.md]: The trailing `.html` should be omitted: `/docs/api/resources/oauth2`

File: _source/_docs/api/resources/oidc.md [hunk 2]
```diff
@@ -1,189 +1,387 @@
...
 
-### Get User Information
+`https://{yourOktaDomain}.com/oauth2/default/v1/authorize`
+
+### /authorize
 {:.api .api-operation}
 
-{% api_operation get /oauth2/v1/userinfo %}
-{% api_operation post /oauth2/v1/userinfo %}
+{% api_operation get ${baseUrl}/v1/authorize %}
```
Comment (jmelberg-okta) [inline _source/_docs/api/resources/oidc.md]: Is the plan to update all of our API operation tags with`${baseUrl}`? If so, why choose this over `{yourOktaDomain}`?

Comment (jakubvul) [reply _source/_docs/api/resources/oidc.md]: Only this page. Explained here: https://developer.okta.com/docs/api/resources/oidc#composing-your-base-url

**Comment by nbarbettini** [comment]
🙌

**Pull request merged**